### DISCUSSION TITLE
Need testers to test this version (Enderpearl bug)

### DISCUSSION THREAD

**Discussion by GoodestEnglish** [diamond-rip/Eden#135] [Announcements]
Currently, for unknown reason, enderpearl throw action will somehow gets cancelled even when the player isn't in cooldown ([video](https://www.youtube.com/watch?v=sduhW6voqLo)). This thread will post development snapshots of the plugin which attempt to fix this bug. Please let me know if the bug still exists or not. Detailed reports are much appreciated. Report examples are like:
- How many pearls you've thrown overall? 
- How many pearls you've thrown which is glitched?
- What did you do before you throw the enderpearl?
- Your guess of why the bug occurs
- etc..

Please use the reply feature when you are comment your reply instead of commenting.

**Comment by GoodestEnglish** [comment]
Attempt to fix the bug by clearing all cooldown when match starts: https://drive.google.com/file/d/1H5wysVjNVntc47_Q3kgo3fjDS7g6Lg-m/view?usp=sharing

**Reply by DevSkript** [reply]
still dont work
>16pearls
>5pearls is glitched
>i dont do anything just stand there and throw to test
>well maybe the timer was a lil bit too fast because when the pearl glitched i dont lost the pearls at all

**Reply by GoodestEnglish** [reply]
> still dont work
> 
> > 16pearls
> > 5pearls is glitched
> > i dont do anything just stand there and throw to test
> > well maybe the timer was a lil bit too fast because when the pearl glitched i dont lost the pearls at all

Thanks for the test. I might going to recode the whole timer system in the coming days. This will take some time since I am having a midterm test in the coming week.

**Comment by GoodestEnglish** [comment]
Reworked part of the cooldown system: https://github.com/diamond-rip/Eden/releases/tag/1.0.37-BETA-b3dcdbf

**Comment by Skipsqueak514** [comment]
the plugin is super good i dont know how this is even free wtf

**Reply by GoodestEnglish** [reply]
Thanks. Im glad you like it ;)

**Reply by Skipsqueak514** [reply]
You have a discord server? If not why not create one, would be cool!

**Reply by GoodestEnglish** [reply]
I have one, but it is used to communicate with my audience. Currently there's no discord server support for Eden. You can either email me, or open a new discussion in https://github.com/diamond-rip/Eden/discussions/new/choose. My email is goodestenglish@diamond.rip
